lib/referee.test.js

```diff
@@ -77,29 +78,25 @@ describe("API", function() {
     // this prevents accidental expansions of the public API
     it("should only have expected properties", function() {
         var expectedProperties = JSON.stringify([
+            "_events",
+            "_eventsCount",
+            "_maxListeners",
             "add",
             "assert",
-            "bind",
             "captureException",
-            "emit",
```

Comment:
This shows that the test case isn't testing the right thing anymore. The original intend was to have a test that verifies the exposed API. Since the `EventEmitter` APIs are now up in the prototype chain, they're not returned by `Object.keys`. I thing the test should collect all exposed functions with a for-in loop.

Reply:
That would mean that each time EventEmitter adds new features the tests will break here, hence the instanceof assertion below. It could also have different props based on which node version this is running in. I don't mind making the change - just making sure it is what you really want?

Reply:
I also even pondered ignoring the props that start with an underscore too.

Reply:
Yes, I agree that it's not ideal to depend on externally managed APIs in the tests. Let's leave it with the instanceof check. The same argument is true for internals like the underscore properties. Shall we check for API functions then?

Reply:
OK, I'll update the assertion to exclude the underscored properties and I'll add a test for explicit existence of `on`/`off, `addEventListener`/`removeEventListener`?

Reply:
Sorry, I'm having a hard time expressing my thoughts clearly today. Basically we just want this test to assert that the API surface doesn't change unexpectedly. I think one way to achieve this could be to find all functions that `referee` directly implements. This list of function names could be compared to the `expectedProperties`.

Reply:
OK, excluded the properties which start with an underscore: https://github.com/sinonjs/referee/pull/210/commits/7d444bcd870196a2e11132d5b29a99b1bb9b2dda

Otherwise the test already does what it says, right? It takes all the exposed object keys and makes sure that they match an exact list?